### PR TITLE
Aa/melgan forward tacotron att

### PR DESCRIPTION
**Pull request by a-a-egorovich**
AC and models description for modified ForwardTacotron and MelGAN pipeline.

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins ok to test

File: models/intel/text-to-speech-en-0001/accuracy-check.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+      datasets:
+        - name: forward_tacotron_melgan_io
+          data_source: forward_tacotron_melgan_io
+          reader: numpy_dict_reader
+          annotation_conversion:
+            converter: multi_feature_regression
+            data_dir: forward_tacotron_melgan_io
+            input_suffix: duration_prediction_in_
+            reference_suffix: melgan_out_
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/accuracy-check.yml:34]: could you please move dataset specific things to data/dataset_definitions.yml and keep only name in config?

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+The forward-tacotron-duration-att-0001 model is a ForwardTacotron-based duration predictor for symbols.
+
+| Metric                                        | Value     |
+|-----------------------------------------------|-----------|
+| GFlops                                        | 15.84      |
+| MParams                                       | 13.569     |
+
+### Inputs
+
+Sequence, name: `input_seq`, shape: [1x512], format: [BxC]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:27]:
```suggestion
Sequence, name: `input_seq`, shape: `1, 512`, format: `B,C`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - C - number of symbols in sequence
+
+Sequence, name: `input_mask`, shape: [1x1x512], format: [BxDxC]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:32]:
```suggestion
Sequence, name: `input_mask`, shape: `1, 1, 512`, format: `B, D, C`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - D - extra dimension for multiplication
+   - C - number of symbols in sequence
+
+Mask for input sequence, name: `input_mask`, shape: [1x1x512], format: [BxDxC]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:38]:
```suggestion
Mask for input sequence, name: `input_mask`, shape: `1, 1, 512`, format: `B, D, C`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - D - extra dimension for multiplication
+   - C - number of symbols in sequence
+
+Mask for relative position representation in attention, name: `pos_mask`, shape: [1x1x512x512], format: [BxDxCxC]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:44]:
```suggestion
Mask for relative position representation in attention, name: `pos_mask`, shape: `1, 1, 512, 512`, format: `B, D, C, C`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - D - extra dimension for multiplication
+   - C - number of symbols in sequence
+
+### Outputs
+
+1. Duration for input symbols, name: `duration`, shape: [1, 512, 1], format [BxCxH]. Contains predicted duration for each of the symbol in sequence.
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:52]:
```suggestion
1. Duration for input symbols, name: `duration`, shape: `1, 512, 1`, format `B, C, H`. Contains predicted duration for each of the symbol in sequence.
```

File: models/intel/index.md [hunk 7]
```diff
@@ -265,6 +265,16 @@ Deep Learning compressed models
...
+## Text To Speech
+
+Deep Learning models for speech synthesis (mel spectrogram generation and wave form generation).
+
+| Model Name                                                                                | Complexity (GFLOPs)  | Size (Mp)  |
+|-------------------------------------------------------------------------------------------|--------------------- |----------- |
+| [text-to-speech-en-0001](./text-to-speech-en-0001/description/text-to-speech-en-0001.md)  |                      |            |
+| forward-tacotron-duration-att-0001                                                        | 15.84                | 13.569     |
+| forward-tacotron-regression-att-0001                                                      | 7.65                 | 4.96       |
+| melgan-upsample-0001                                                                      | 48.38                | 12.77      |
```
Comment (IRDonch) [inline models/intel/index.md:277]: The names of the components of a composite model should start with the name of that composite model.

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: models/intel/index.md (hunk 7 above)
Comment (a-a-egorovich) [reply models/intel/index.md:277]: Like this:
text-to-speech-en-0001-forward-tacotron-duration-att
text-to-speech-en-0001-forward-tacotron-regression-att
text-to-speech-en-0001-melgan-upsample

?

Comment (IRDonch) [reply models/intel/index.md:277]: That's better, but it seems unnecessarily verbose. Component names are generally supposed to describe the function of that particular component. For the first two it seems most logical to use the same names as in `forward-tacotron`:

* `text-to-speech-en-0001-duration-prediction`
* `text-to-speech-en-0001-regression`

I don't know what the best way to describe the last component is (`generation`, maybe?), but it should be something similar.

File: models/intel/text-to-speech-en-0001/accuracy-check.yml (hunk 1 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/accuracy-check.yml]: done

File: models/intel/index.md (hunk 7 above)
Comment (a-a-egorovich) [reply models/intel/index.md]: Done

**Review by IRDonch — APPROVED** [review]

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+   - D - extra dimension for multiplication
+   - C - number of symbols in sequence
+
+### Outputs
+
+1. Duration for input symbols, name: `duration`, shape: `1, 512, 1`, format `B, C, H`. Contains predicted duration for each of the symbol in sequence.
+   - B - batch size
+   - C - number of symbols in sequence
+   - H - empty dimension
+2. Processed embeddings, name: `embeddings`, shape: [1, 512, 256], format [BxCxH]. Contains processed embeddings for each symbol in sequence.
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:56]: on this line I'm tired of correcting shape and layout formating, but It should be understandable that it should be the same for all places. Could you please align?

Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:56]:
```suggestion
2. Processed embeddings, name: `embeddings`, shape: `1, 512, 256`, format `B, C, H`. Contains processed embeddings for each symbol in sequence.
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+| Metric                                        | Value     |
+|-----------------------------------------------|-----------|
+| GFlops                                        | 7.65      |
+| MParams                                       | 4.96      |
+
+
+### Inputs
+
+Processed embeddigs aligned by durations, name: `data`, shape: [1x512x256], format: [BxTxC]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:73]:
```suggestion
Processed embeddigs aligned by durations, name: `data`, shape: `1, 512, 256`, format: `B, T, C`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - T - time in mel-spectrogram
+   - C - processed embedding dimension
+
+Mask for 'data' by time dimension, name: `data_mask`, shape: [1x1x512], format: [BxDxT]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:79]:
```suggestion
Mask for 'data' by time dimension, name: `data_mask`, shape: `1, 1, 512`, format: `B, D, T`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - D - extra dimension for multiplication
+   - T - time in mel-spectrogram
+
+Mask for relative position representation in attention, name: `pos_mask`, shape: [1x1x512x512], format: [BxDxCxC]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:85]:
```suggestion
Mask for relative position representation in attention, name: `pos_mask`, shape: `1, 1, 512, 512`, format: `B, D, C, C`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - D - extra dimension for multiplication
+   - C - number of symbols in sequence
+
+### Output
+
+Mel-spectrogram, name: `mel`, shape: [80x512], format: [CxT]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:93]:
```suggestion
Mel-spectrogram, name: `mel`, shape: `80, 512`, format: `C, T`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+
+| Metric                                        | Value |
+|-----------------------------------------------|-------|
+| GFlops                                        | 48.38 |
+| MParams                                       | 12.77 |
+
+
+### Inputs
+
+Mel-spectrogram, name: `mel`, shape: [1x80x128], format: [BxCxT]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:111]:
```suggestion
Mel-spectrogram, name: `mel`, shape: `1, 80, 128`, format: `B, C, T`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+where:
+   - B - batch size
+   - C - number of rows in mel-spectrogram
+   - T - time in mel-spectrogram
+
+### Outputs
+
+Audio, name: `audio`, shape: [32768], format: [T]
```
Comment (eaidova) [inline models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md:119]:
```suggestion
Audio, name: `audio`, shape: `32768`, format: `T`
```

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md (hunk 8 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md (hunk 9 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md (hunk 10 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md (hunk 11 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md (hunk 12 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md (hunk 13 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

File: models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md (hunk 14 above)
Comment (a-a-egorovich) [reply models/intel/text-to-speech-en-0001/description/text-to-speech-en-0001.md]: Done

**Review by eaidova — APPROVED** [review]

**Pull request merged**